### PR TITLE
3624 Add kwargs to torch APIs of utilities

### LINKED ISSUE #3624: Add kwargs to pytorch_numpy_unification
**Is your feature request related to a problem? Please describe.**
As we already added `kwargs` to the `ascontiguousarray` utility, I checked all the other utilities in `pytorch_numpy_unification`, these items also have useful additional args of PyTorch Tensor:
https://pytorch.org/docs/stable/generated/torch.quantile.html?highlight=torch%20quantile#torch.quantile
https://pytorch.org/docs/stable/generated/torch.cumsum.html?highlight=torch%20cumsum#torch.cumsum
https://pytorch.org/docs/stable/generated/torch.searchsorted.html?highlight=torch%20searchsorted#torch.searchsorted
https://pytorch.org/docs/stable/generated/torch.repeat_interleave.html?highlight=torch%20repeat_interleave#torch.repeat_interleave
Would be nice to also add `kwargs` for them.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3624 .

### Description
This PR added the `kwargs` to torch utilities as `contiguous`, and also added doc-strings.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utils_pytorch_numpy_unification.py [hunk 1]
```diff
@@ -112,7 +114,7 @@ def percentile(x: NdarrayOrTensor, q, dim: Optional[int] = None) -> Union[Ndarra
     else:
         q = torch.tensor(q, device=x.device)
         if hasattr(torch, "quantile"):  # `quantile` is new in torch 1.7.0
-            result = torch.quantile(x, q / 100.0, dim=dim)
+            result = torch.quantile(x, q / 100.0, dim=dim, **kwargs)
```
Comment (wyli) [inline monai/transforms/utils_pytorch_numpy_unification.py:117]: `np.percentile` also takes some additional params, perhaps we try to support both, like https://github.com/Project-MONAI/MONAI/blob/3c6ee694d5e7fd6d9da2c65ed1b83030488da064/monai/transforms/utils_pytorch_numpy_unification.py#L301-L305

Comment (Nic-Ma) [reply monai/transforms/utils_pytorch_numpy_unification.py]: Hi @wyli ,

Thanks for your suggestion.
In this case, I found that `numpy` has more args than `tensor`, so I added `keepdim` to the function and use `kwargs` for numpy array.

Thanks.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**